### PR TITLE
Publishing concept

### PR DESCRIPTION
**Pull request by bkuhl**
Hasn't been tested end to end yet, but the domain logic is there

### CODE REVIEW

**Comment by bkuhl** [comment]
Pushed up a refactor that supports a single endpoint for all sources (GitHub, GitLab, etc.) and all events, using a factory to delegate work.

**Comment by millidavids** [comment]
+1

**Comment by jlaswell** [comment]
Still reviewing here. Please don't merge.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,13 @@
-## NPM Publisher [![Build Status](https://travis-ci.org/realpage/npm-publisher.svg?branch=master)](https://travis-ci.org/realpage/npm-publisher)
+## Asset Publisher [![Build Status](https://travis-ci.org/realpage/asset-publisher.svg?branch=master)](https://travis-ci.org/realpage/asset-publisher)
+
+This service responds to GitHub web hooks looking for semantically versioned tags and deploying minor and patch versions to a distributor (i.e. - Amazon S3).
+
+ * `PRIVATE_KEY` - Public pair
```
Comment (jlaswell) [inline README.md]: What do you mean by 'Public pair'? Just the private key from the key pair?

File: app/VCS/Git.php [hunk 2]
```diff
@@ -2,23 +2,45 @@
...
+        $process = app(Process::class, ['git checkout tags/'.$version->minorTag()]);
+        $process->setWorkingDirectory($repository->path());
         $process->run();
 
         if (!$process->isSuccessful()) {
             throw new ProcessFailedException($process);
         }
 
-        return $path.DIRECTORY_SEPARATOR.$repository->namespace().'/'.$repository->name();
+        return true;
```
Comment (jlaswell) [inline app/VCS/Git.php]: I think we need to clean up the temp path. Am I missing where that is happening?

File: composer.json [hunk 3]
```diff
@@ -9,11 +9,13 @@
         "laravel/lumen-framework": "5.2.*",
         "vlucas/phpdotenv": "~2.2",
         "graham-campbell/flysystem": "^3.0",
-        "league/flysystem-aws-s3-v3": "~1.0"
+        "league/flysystem-aws-s3-v3": "~1.0",
+        "mockery/mockery": "dev-master"
```
Comment (jlaswell) [inline composer.json]: Should be in "require-dev"

File: README.md (hunk 1 above)
Comment (bkuhl) [inline README.md]: This is part of a WIP on the docker portion of this image.  I'll remove that.

File: app/VCS/Git.php (hunk 2 above)
Comment (bkuhl) [inline app/VCS/Git.php]: Can you define "clean up"?

File: composer.json (hunk 3 above)
Comment (bkuhl) [inline composer.json]: 👍 Fixed

File: app/VCS/Git.php (hunk 2 above)
Comment (jlaswell) [inline app/VCS/Git.php]: Just deleting it.

Comment (bkuhl) [inline app/VCS/Git.php]: Sorry for the confusion on my end.  What line(s) are you suggesting we delete?

Comment (jlaswell) [inline app/VCS/Git.php]: The path created in the [temporaryPath()](https://github.com/realpage/asset-publisher/pull/1/files#diff-8b4360535d2feb67c65fed2457b0efedR44) method.

Comment (bkuhl) [inline app/VCS/Git.php]: The value of that method is that it provides a temporary, unique directory for the code to be checked out in.  In the event of failure we won't have to worry about any collisions.

Comment (jlaswell) [inline app/VCS/Git.php]: I was looking for where the directory is deleted, but I [found it](https://github.com/realpage/asset-publisher/pull/1/files#diff-c5a689eaabc4acf77aac709d4993225aR41).

**Comment by jlaswell** [comment]
lgtm

File: app/VCS/Git.php (hunk 2 above)
Comment (bkuhl) [inline app/VCS/Git.php]: Your bringing this up did make me realize that this directory isn't cleaned up in the event of an exception, so I'm going to improve that in the next PR

**Pull request merged**